internal-services/catalog/create-advisory-task.yaml

```diff
@@ -81,35 +81,18 @@ spec:
           gitlab_init
           git_functions_init
 
-          function get_advisory_num {
-              # This function will skip many advisory numbers as it is so tied to date. Need to
-              # discover if skipping advisory numbers like this is okay.
-              # There is also a race condition picking the advisory number like this.
-              EXISTING=$(for advisory in $(find data/advisories/*/${1} -name "advisory.yaml") ; do
-                  yq '.metadata.name' $advisory | grep ":${2}" || true
-              done | sort -r | head -n 1)
-              if [[ ! -z "$EXISTING" ]] ; then
-                  echo "${2}$((${EXISTING:9}+1))" # Strip first 9 chars for YYYY:MMDD then add 1
-                  return
-              fi
-              echo "${2}0" # First advisory of the day
-          }
-
           # This also cds into the git repo
           git_clone_and_checkout --repository "$REPO" --revision "$REPO_BRANCH"
 
-          YEAR=$(date '+%Y')
-          ADVISORY_DIR="data/advisories/$(params.origin)/${YEAR}" # group advisories by <origin workspace>/year
-          ADVISORY_NUM=$(get_advisory_num $YEAR $(date '+%m%d'))
-          ADVISORY_NAME="${YEAR}:${ADVISORY_NUM}"
-          ADVISORY_FILEPATH="${ADVISORY_DIR}/${ADVISORY_NUM}/advisory.yaml"
-          mkdir -p "${ADVISORY_DIR}/${ADVISORY_NUM}"
+          ADVISORY_DIR="staging/advisories/$(params.origin)/$(date '+%Y')" # group advisories by <origin workspace>/year
+          mkdir -p "${ADVISORY_DIR}"
+          ADVISORY_FILEPATH="${ADVISORY_DIR}/$(uuidgen).yaml"
 
           # Create advisory file
           /home/utils/apply_template.py -o $ADVISORY_FILEPATH --template /home/templates/advisory.yaml.jinja \
-          --data '{"advisory_name":"'$ADVISORY_NAME'","advisory":$(params.advisory_json)}'
+          --data '{"advisory":$(params.advisory_json)}'
```

Comment:
I assume `advisory_name` will be injected by the Gitlab CI pipeline that will generate the name, right?

Reply:
yep. The CI pipeline will move it from staging to data and set the advisory name (both the filename and `metadata.name`)

Reply:
Yep, it will. The name needs to have a numeration that increases by 1 with each new advisory, so it needs to be done in the CI itself to avoid race conditions.